### PR TITLE
Make CLI spend e-cash more configurable

### LINKED ISSUE #3834: `fedimint-cli spend` should have a flag to give exact change or error if it can't
Currently `fedimint-cli spend <amount>` will give us `<amount>` if it has those notes on hand, and give more than we ask for if it doesn't.

I was just writing a test and this behavior caused my test to be flaky. It would be nice for this reason (and probably in general) to be able to request exact amounts from `fedimint-cli spend`.

### PR DESCRIPTION
**Pull request by elsirion**
Fixes #3834

### CODE REVIEW

File: fedimint-cli/src/client.rs [hunk 1]
```diff
@@ -57,8 +59,17 @@ pub enum ClientCmd {
...
+        /// The amount of e-cash to spend
         #[clap(value_parser = parse_fedimint_amount)]
         amount: Amount,
+        /// If the exact amount cannot be represented, return e-cash of a higher
+        /// value instead of failing
+        #[clap(long)]
+        allow_overpay: bool,
+        /// After how many seconds we will try to reclaim the e-cash if it
+        /// hasn't been redeemed by the recipient
+        #[clap(long, default_value_t = 3600)]
```
Comment (dpc) [inline fedimint-cli/src/client.rs:71]: **thought:**  Is this automatic behavior? If so, isn't this a bit low for a default for something that might be passed in a DM or even piece of paper?

**Review by dpc — APPROVED** [review]

File: fedimint-cli/src/client.rs (hunk 1 above)
Comment (elsirion) [reply fedimint-cli/src/client.rs:71]: Good point, what would you propose? A week?

Comment (dpc) [reply fedimint-cli/src/client.rs:71]: Sounds reasonable.

BTW. Maybe we should also `eprintln!` the deadline to warn/educate the user?

Comment (elsirion) [reply fedimint-cli/src/client.rs:71]: could that be a `warn` level log? These are still shown and should be printed to stderr anyway (if not we should fix that).

Comment (dpc) [reply fedimint-cli/src/client.rs:71]: The way I think about it is: logging is for reporting what the program is doing, while `(e)println` for the actual output for the user. Not feeling too strongly about it one way or another.

Comment (elsirion) [reply fedimint-cli/src/client.rs:71]: We need a third stream, stdlog! :laughing:

Comment (justinmoon) [reply fedimint-cli/src/client.rs:71]: In the CLI what would trigger this refund?

Comment (elsirion) [reply fedimint-cli/src/client.rs:71]: Yep, basically anything that runs the SM executor for long enough. What I like to do to trigger this manually is create a LN invoice I don't intend to pay and then await it. We might want a more general purpose "run executor" command.

**Review by douglaz — APPROVED** [review]
Requires rebase, but LGTM

**Review by Maan2003 — APPROVED** [review]

**Pull request merged**